src/molecular-weight.jl

```diff
@@ -0,0 +1,105 @@
+# AA_N   # Asparagine                  ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/6267
+# AA_D   # Aspartate                   ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/5960
+# AA_C   # Cysteine                    ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/5862
+# AA_Q   # Glutamine                   ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/5961
+# AA_E   # Glutamate                   ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/33032
+# AA_G   # Glycine                     ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/750
+# AA_H   # Histidine                   ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/6274
+# AA_I   # Isoleucine                  ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/6306
+# AA_L   # Leucine                     ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/6106
+# AA_K   # Lysine                      ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/5962
+# AA_M   # Methionine                  ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/6137
+# AA_F   # Phenylalanine               ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/6140 
+# AA_P   # Proline                     ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/145742
+# AA_S   # Serine                      ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/5951
+# AA_T   # Threonine                   ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/6288
+# AA_W   # Tryptophan                  ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/6305
+# AA_Y   # Tyrosine                    ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/6057
+# AA_V   # Valine                      ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/6287
+# AA_O   # Pyrrolysine                 ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/21873141
+# AA_U   # Selenocysteine              ->  Source = https://pubchem.ncbi.nlm.nih.gov/compound/25076
+
+# Source of the used water molecular weight = 18.02 gmol for calculations
+# Water                                ->  Source = https://www.cmu.edu/gelfand/k12-educational-resources/polymers/what-is-polymer/molecular-weight-calculation.html 
+
+# Creating the array AA_WEIGHTS of length 28 to list all weights. In ambigous cases, let's list the string "Undefined", so if called, it returns an error
+AA_WEIGHTS = [89.09, 174.20, 132.12, 133.10, 121.16, 146.14, 147.13, 75.07, 155.15, 131.17, 131.17, 146.19, 149.21, 165.19, 115.13, 105.09, 119.12, 204.22, 181.19, 117.15, 255.31, 168.06, "Undefined", 131.17, "Undefined", "Undefined", 0, 0]
+
+# Defining the function molecular_weight, which accepts an amino acid sequence and return molecular weight in  g/mol. Note that the function also account for lost of water for each peptide bond formation 
+function molecular_weight(aa_seq::AASeq)
+    weight = 0
```

Comment:
Since the array loads `Float64` values, it's best to begin the iteration with `0.0` (a float) instead of `0` (an integer). Probably the compiler will take care of it, but let's not rely on the compiler being too clever.